libfoundation/src/system-library-mac.hpp

```diff
@@ -136,19 +149,23 @@ class __MCSLibraryHandleMac: public __MCSLibraryHandlePosix
      * framework. A path matching this pattern is transformed to:
      *    <folder>/<leaf>.framework/<leaf>
      * So that we can use dlopen's default search behavior. */
-    static bool
-    ResolveFrameworkExecutable(MCStringRef p_native_path,
-                               MCStringRef& r_exe)
+    bool
+    ResolveFrameworkExecutable(MCStringRef p_native_path, bool p_has_extension)
     {
         const char *t_framework_ext = ".framework";
         
-        if (!MCStringEndsWithCString(p_native_path,
+        uindex_t t_last_component_end = MCStringGetLength(p_native_path);
+        if (MCStringEndsWithCString(p_native_path,
                                      reinterpret_cast<const char_t *>(t_framework_ext),
                                      kMCStringOptionCompareCaseless))
+        {
+            t_last_component_end -= strlen(t_framework_ext);
+        }
+        else if (p_has_extension)
         {
             return false;
```

Comment:
@runrevmark if there is an explicit extension and it's not framework the we back out here

Reply:
Right - but if there is no explicit extension and you have 'foo' and 'foo.framework', the framework codepath will trigger and you'll only ever bind to foo.framework - never foo.

Reply:
The order of trying things could be inverted, but then I'd get 'foo.framework' if 'foo' isn't there - which might not be what you want. (Again, this is only a concern because this is meant to be a thin wrapper around system functionality).